control/plugin/execution.go

```diff
@@ -115,21 +115,37 @@ func (e *ExecutablePlugin) Run(timeout time.Duration) (Response, error) {
 	e.cmd.Start()
 	e.captureStderr()
 	go func() {
-		for stdOutScanner.Scan() {
-			// The first chunk from the scanner is the plugin's response to the
-			// handshake.  Once we've received that, we can begin to forward
-			// logs on to snapd's log.
-			if !respReceived {
-				respBytes := stdOutScanner.Bytes()
-				err = json.Unmarshal(respBytes, &resp)
-				respReceived = true
-				close(doneChan)
-			} else {
-				execLogger.WithFields(log.Fields{
-					"plugin": path.Base(e.cmd.Path()),
-					"io":     "stdout",
-				}).Debug(stdOutScanner.Text())
+		for {
+			for stdOutScanner.Scan() {
+				// The first chunk from the scanner is the plugin's response to the
+				// handshake.  Once we've received that, we can begin to forward
+				// logs on to snapd's log.
+				if !respReceived {
+					respBytes := stdOutScanner.Bytes()
+					err = json.Unmarshal(respBytes, &resp)
+					respReceived = true
+					close(doneChan)
+				} else {
+					execLogger.WithFields(log.Fields{
+						"plugin": path.Base(e.cmd.Path()),
+						"io":     "stdout",
+					}).Debug(stdOutScanner.Text())
+				}
 			}
+
+			if err := stdOutScanner.Err(); err != nil {
+				reader := bufio.NewReader(e.stdout)
+				log, err := reader.ReadString('\n')
```

Comment:
Same comments made below on the stderr routine apply here (re: handle ReadString error).